### ISSUE TITLE
Doctest syntax highlighting does not work with `#[doc]` attributes

### ISSUE THREAD

**Issue by c410-f3r** [rust-analyzer/rust-analyzer#5016]
`cfg` and friends paired with `doc = "..."` should trigger syntax highlight if applicable.

```rust
/// # Example
///
/// ```rust
/// let _ = example(&[1, 2, 3]);
/// ```
///
/// # Another example
///
#[cfg_attr(feature = "alloc", doc = "```rust")]
#[cfg_attr(not(feature = "alloc"), doc = "```ignore")]
/// let _ = example(&alloc::vec![1, 2, 3]);
/// ```
pub fn example<T>(t: &T) -> &[i32]
where
    T: AsRef<[i32]>
{
  t.as_ref()
}
```
